scripts/test-util.sh

```diff
@@ -179,3 +179,76 @@ download_ovmf() {
     time wget --quiet $OVMF_FW_URL || exit 1
     popd || exit
 }
+
+# Function to mount image partition, execute commands, and cleanup.
+# Arguments: $1: Image file path, $2: Mount directory, $3+: Commands to execute.
+mount_and_exec() {
+    local IMG="$1"
+    local MOUNT_DIR="$2"
+    local LOOP_DEV=""
+    local PARTITION_DEV=""
+    local COMMAND_STATUS=0
+
+    if [ ! -f "$IMG" ] || [ -z "$MOUNT_DIR" ]; then return 1; fi
+    mkdir -p "$MOUNT_DIR"
+
+    # Create loop device for the entire disk image
+    LOOP_DEV=$(sudo losetup -f --show "$IMG")
+    if [ -z "$LOOP_DEV" ]; then return 1; fi
+
+    # Scan for partitions and define partition device node (p1)
+    sudo partprobe "$LOOP_DEV" 2>/dev/null
+    PARTITION_DEV="${LOOP_DEV}p1"
+
+    # Wait for partition node
+    if ! sudo test -b "$PARTITION_DEV"; then
+        sleep 1
+        if ! sudo test -b "$PARTITION_DEV"; then
+            sudo losetup -d "$LOOP_DEV" || true
+            return 1
+        fi
+    fi
+
+    # Mount the partition
+    if ! sudo mount "$PARTITION_DEV" "$MOUNT_DIR"; then
+        sudo losetup -d "$LOOP_DEV" || true
+        return 1
+    fi
+
+    # Execute the commands
+    shift 2
+    "$@"
+    COMMAND_STATUS=$?
+
+    # Unmount and Detach
+    sudo umount "$MOUNT_DIR" || true
+    sudo losetup -d "$LOOP_DEV" || true
```

Comment:
Please use a trap to unmount and detach the loop device on EXIT (better signals too, say `INT TERM`).